ui/src/reusable_ui/components/dropdowns/Dropdown.tsx

```diff
@@ -0,0 +1,174 @@
+import React, {Component, CSSProperties, Fragment} from 'react'
+import classnames from 'classnames'
+import uuid from 'uuid'
+import _ from 'lodash'
+
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import {
+  ComponentColor,
+  ComponentSize,
+  IconFont,
+  DropdownMenuColor,
+} from 'src/reusable_ui/types'
+import DropdownButton from 'src/reusable_ui/components/dropdowns/DropdownButton'
+import FancyScrollbar from 'src/shared/components/FancyScrollbar'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import './Dropdown.scss'
+
+interface Props {
```

Comment:
why are there so many optional props? is this a smell of something else?

Reply:
I would say it also reflects the modular styles used in the atomic components